Tweaked tests

Lagicrus opened the pull request:
Updated the tests for the Dupe test on the suggestion by @ear1grey and @matt-thepie

Lagicrus committed:
main

Lagicrus committed:
updated tests

Lagicrus committed:
minor update

Lagicrus committed:
Revert "minor update"

This reverts commit 3008a15ee64e241a4885bc9656ad660201352cc4.

Lagicrus committed:
Revert "updated tests"

This reverts commit 4b7c02686c0282fd1fc8a05a914f9a318fb24fb4.

Lagicrus committed:
Revert "main"

This reverts commit ecc5ead5b8f43d4119377a22d20f7ce3b8b7c5e6.

Lagicrus committed:
Updated tests for the dupe test

Added code to do more indepth checking of the two elements to make sure that they are inface the same as before you can just trick it with copying text content across

Lagicrus commented:
True, the reason I was testing for this was due to if you use [cloneNode()](https://developer.mozilla.org/en-US/docs/Web/API/Node/cloneNode) the ID is duplicated so testing this shows that cloneNode() was used / the user did a "better" job of cloning it. 
The other reason was that from what I understood from talking with Matt and Rich was that it was expected within this test that we cloned the element and if the user did their "job" then this would also be copied across. 
As a final thought, if you did check for the ID is the same you could bring up a warning saying that in practice you don't want to copy the ID. Or alternatively, you could have the user get the element by class name and then check for that instead as that isn't against guidelines (to my understanding).

jacekkopecky commented:
You're right that cloneNode() copies the ID. It's not hard to strip the ID afterwards. I'd suggest that you flip that part of the test and expect the IDs (if any) *not* to be the same.

Rich and Matt can chime in if they'd prefer a different approach.

Lagicrus committed:
Making sure the IDs are not the same

Made sure that the IDs are not the same along with a explanation as to what to use instead if you have to duplicate attributes

Lagicrus commented:
Swapped from using `.strictEqual` to `.notEqual` and added a warning as to what you should use instead of an ID if you wanted to duplicate the attributes across from one element to another.

jacekkopecky commented on assess.dom.js:
```diff
@@ -519,6 +519,25 @@ QUnit.test("duplication",
+            "The two elements should have children"
+        );
+
+        assert.propEqual(
```
I just noticed the use of `propEqual` - are you sure this does what you think it does? You might instead compare the innerHTML; then the above assertion with `hasChildNodes()` is not necessary either, or the `textContent` comparison above.

jacekkopecky commented on assess.dom.js:
```diff
@@ -519,6 +519,25 @@ QUnit.test("duplication",
 
+        assert.notEqual(
+            paras[0].id,
+            paras[1].id,
```
try something like `assert.ok((paras[0].id == null && paras[1].id == null) || paras[0].id != paras[1].id, ...`

this would be more robust if in the future we change the selection code to not use an ID, and then the element to be duplicated wouldn't even have an ID.

Lagicrus replied:
Resolved in https://github.com/portsoc/ws_dom/pull/4/commits/6e8a53d851184c296e271eb8b8aeb76a0eea85e5

Lagicrus committed:
Replaced notEqual with ok

Replaced the notEqual code for checking IDs with code that passes if they are both null OR if they are not the same.
